metrics/zio-telemetry-open-telemetry-backend/src/main/scala/sttp/client3/ziotelemetry/opentelemetry/ZioTelemetryOpenTelemetryBackend.scala

```diff
@@ -0,0 +1,52 @@
+package sttp.client3.ziotelemetry.opentelemetry
+
+import io.opentelemetry.api.trace.SpanKind
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator
+import io.opentelemetry.context.propagation.{TextMapPropagator, TextMapSetter}
+import sttp.capabilities.Effect
+import sttp.client3._
+import sttp.client3.impl.zio.ExtendEnv
+import zio._
+import zio.telemetry.opentelemetry.TracingSyntax.OpenTelemetryZioOps
+import zio.telemetry.opentelemetry._
+
+import scala.collection.mutable
+
+class ZioTelemetryOpenTelemetryBackend[+P] private (
+    delegate: SttpBackend[RIO[Tracing, *], P],
+    tracer: ZioTelemetryOpenTelemetryTracer
+) extends DelegateSttpBackend[RIO[Tracing, *], P](delegate) {
+  def send[T, R >: P with Effect[RIO[Tracing, *]]](request: Request[T, R]): RIO[Tracing, Response[T]] = {
+    val carrier: mutable.Map[String, String] = mutable.Map().empty
```

Comment:
is it safe to use an ordinary mutable map here? can't it be updated from multiple threads concurrently?

Reply:
It's modified only in [W3CTraceContextPropagator](https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/trace/propagation/W3CTraceContextPropagator.java#L133,L147) `inject` method, which is called only once, so I think it should be safe.